### ISSUE TITLE
Copy mode: cannot move within source

### ISSUE THREAD

**Issue by fabien** [bevacqua/dragula#23]
I believe this line https://github.com/bevacqua/dragula/blob/master/dragula.js#L257 prevents moving (reordering) items within the source container, when copy is `true`.

I think it should be possible to still reorder items within the same container.

**Comment by fabien** [comment]
Here's a workaround that seems to work pretty well. I wanted the following behavior:
- left is the 'target' zone
- right is the 'source' zone, containing all of the available items (copied)
- you cannot move items from left to right
- you can only copy items from right to left
- you can reorder items in left
- you cannot reorder items in right

```
dragula([left, right], { 
  copy: true,
  moves: function (el, container) {
    return el.parentElement === right;
  }
});

dragula([left], { 
  removeOnSpill: true
});
```

**Comment by bevacqua** [comment]
Feature, not a bug. Most of the time you don't want to allow reordering in copy scenarios

**Comment by bravedick** [comment]
+1 for ordering while copy. Example:

Email builder.

Left side - unique template blocks: text, logo, image, tabe ...
Right side - email. You drag templates block here and modify them using some WYSIWYG
Emails can have several text blocks -> we need use copy
Block's order in email can be changed -> we need reordering

If you don't want to implement it in core - it is good to suggest some workaround in docs.

**Comment by Gmattgreenfield** [comment]
Yer, this would be great - for exactly the same reason as mentioned above. 

I would agree that the source should not be order-able but to have it on the drop area would be really handy.

Could there not just be an option to disable reordering? it could work across all of the use cases, not just copy.

Great plugin by the way, thanks for making it!

**Comment by muckinger** [comment]
+1 for ordering while copy

**Comment by robnov** [comment]
+1 for allowing target and/or source ordering during copy (perhaps via an argument). Would really simplify alternative workarounds. Great framework by the way.

**Comment by patrickburke** [comment]
+1 for ordering while copy (maybe with extra configuration?)

In my current scenario the user takes some components from one container, drags them into the other to create a web page. But it should also be possible to reorder them once dragged over.

The workaround mentioned by @fabien works though...

**Comment by bevacqua** [comment]
Problem here is that the current implementation clones the item when drag starts, and the original stays in the container (signaling a copy). If we allowed to reorder on source we'd have to reimplement the UX and I can't figure out how we might do that.

**Comment by patrickburke** [comment]
Hm, seems like a small misunderstanding. What I was talking about was reordering only in target area, not in source. Which is what @fabien and @Gmattgreenfield meant as well, I think.

And know that I think about it, this doesn't even need to be a config setting or something. The workaround mentioned by @fabien is probably the solution! :tada: Maybe put it into docs?

I'll create a PR if I find the time this evening! :muscle:

**Comment by bevacqua** [comment]
"this line prevents moving (reordering) items within the source container, when `copy` is `true`.", I assumed the discussion was about source container.

**Comment by bravedick** [comment]
+1 for reordering in target area.

Something like: moving inside container = ordering, moving outside = copy.

**Comment by patrickburke** [comment]
Well, in his second comment he says he wants to be able to reorder stuff in the target container (left)...

**Comment by bevacqua** [comment]
Got it. This would be resolved by `isCopy` then. #160

**Comment by patrickburke** [comment]
:+1:

**Comment by bevacqua** [comment]
Implemented in `3.1.0`
